### PR TITLE
#19457: emitMultiple should handle empty iterator

### LINKED ISSUE #19457: emitMultiple in GraphStage does not call andThen if iterator is empty
https://github.com/akka/akka/blob/7ee50d18ae9793e21960e6d4d00f6fb3ed0cf4fa/akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala#L760

`emitMultiple` does check `if (element.hasNext)` and if it is false it does nothing but should call `andThen`.

### PR DESCRIPTION
**Pull request by drewhk**
Fixes #19457

### CODE REVIEW

**Comment by johanandren** [comment]
LGTM!

**Comment by rkuhn** [comment]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala [hunk 1]
```diff
@@ -67,6 +68,18 @@ class GraphStageLogicSpec extends AkkaSpec with GraphInterpreterSpecKit with Con
     }
   }
 
+  object emitEmptyIterable extends GraphStage[SourceShape[Int]] {
+    val out = Outlet[Int]("out")
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: for discussion in the future – we have 2 different styles now, sometimes "out" and sometimes "StageName-out", we should settle on one style, (not for this issue ofc)

**Comment by ktoso** [comment]
LGTM, oh Endre merged first :) Good

File: akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala (hunk 1 above)
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: this is just a test class and I was lazy. StageName.out is the official one.

Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: Ah, right, doesnt matter that much in test.

**Pull request merged**